feat(pulsar sink): support tls options (#22148)
vectordotdev/vector: changelog.d/pulsar_sink_source_support_tls_options.feature.md

> Tls options to set custom certificate chain are now available for `pulsar` sink and source.

Reviewer: ```suggestion
The `pulsar` source and sink now support configuration of TLS options via the `tls` configuration field.
```

Cleaning this up a bit.

Author: fixed in https://github.com/vectordotdev/vector/pull/22148/commits/85de10c5985602d1a4cf8408b7adaf355a28db8b
